R/extract.R

```diff
@@ -67,15 +72,22 @@ filter_chunk_end_positions <- function(starts, ends) {
   if (length_difference == 0 && all(ends > starts)) {
     return(ends)
   }
-  if (length_difference < 0) {
-    stop("Malformed file!", call. = FALSE)
-  }
 
   positions <- sort(c(starts = starts, ends = ends))
   code_start_indexes <- grep("starts", names(positions))
-  code_ends <- positions[1 + code_start_indexes]
 
-  stopifnot(all(grepl("ends", names(code_ends))))
+  code_ends <- positions[pmin(1 + code_start_indexes, length(positions))]
+
+  bad_end_indexes <- which(grepl("starts", names(code_ends), fixed = TRUE))
```

Comment:
`which(grepl(.))` is just `grep()` right?

Reply:
Yep, totally overlooked. Thanks!